collections/ffs.go

```diff
@@ -0,0 +1,70 @@
+package collections
+
+import (
+	"context"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+type FFSInstance struct {
+	BucketKey string `bson:"_id"`
+	FFSToken  string
+	Archives  Archives
+}
+
+type Archives struct {
+	Current Archive
+	History []Archive
+}
+
+type Archive struct {
+	Cid        string
```

Comment:
Mongo will bonk if you make this `cid.Cid`?

Reply:
Good question, will try. I suppose that will call `MarshalJSON` sooo it shouldn't be a problem.
Will report back here.

Reply:
Maybe the same goes for some of the other types here... not sure if they are representing some higher type like `Cid`

Reply:
Oh, doesn't work since Mongo allows to provide serialization of BSON:
```
// Marshaler is an interface implemented by types that can marshal themselves
// into a BSON document represented as bytes. The bytes returned must be a valid
// BSON document if the error is nil.
type Marshaler interface {
	MarshalBSON() ([]byte, error)
}
```
So that would need some augmenting type.

But, took advantage and switched to store bytes... sounds that would be better (even in BSON) than storing a base32 string cid.

Reply:
Sounds good. For the other collections, I opted for the much more painful approach of handling the encoding / decoding internally so that the consumer doesn't have to do something extra before using the values. I think that's pretty nice, but I'm not sure if it's worth it in this case.

Reply:
Yep, for thread IDs etc, I've just been storing bytes, and then decoding into the right type on the way out.

Reply:
Well, to make that happen the API can use a struct with `cid.Cid`, and then save internally another struct with `[]bytes`. That would only mean moving that custom `.Bytes()` internally, and still use the automatic encoding/decoding which might be a bit less painful (would be kind of a direct mapping of fields). Mentioning that as another middle-point option.

Reply:
That sounds like a good option. Pretty close to the other collections, but they're using the `bson.M` instead of an additional struct. That struct can carry `bson` tags with JSON looking tags (snakecase), which will make playing with data in the mongo CLI a bit more intuitive since all key patterns would match.